### PR TITLE
Video: Pause video when trimming

### LINKED ISSUE #9108: Pause video while dragging trim handles
## Feature Description

Pause the video while dragging the trim handles at the current frame of the dragged handle, thus scrubbing it backward and forward as the user drags the offset. Once the user releases the handle, start looping the video again.

Do the same things for keyboard input, but with a slight delay before re-starting playing the video (so if the user makes multiple keypresses in a row, keep the video paused at that end, but updated the frame to the current state of the offset, and only once the user has stopped pressing keys for a while (200ms?), restart looping the video.

Of course, don't start the video again if it was paused from the start - but do still move the video play head to the offset that is being updated.

### PR DESCRIPTION
**Pull request by miina**
## Summary
Does the following:
- The video stops while dragging the trim handles and displays the relevant frame.
- The video stops for 1000ms when using the arrow keys for moving the trim handles.
- When the user manually stops the video before using the trim handles, the video won't start playing again after using the trim handles.

## User-facing changes

| Before | After |
|-|-|
| pause-trim-before | pause-trim-after |

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Add a video (preferably such a video where the content changes enough to detect different moments in the video)
2. Enter trim mode, ensure the video is playing
3. Now drag the trim handles. Verify that when dragging the trim handles, the video pauses and will also adjust itself to display the frame of where the handle currently is.
4. Let go of the handle, verify the video start looping in the new trim range.
5. Now try the same thing with keyboard, using arrow keys. Again, verify, the video pauses while using the drag handles and that it adjusts the displayed frame according to the handle. When stopping using the keys, the video starts playing again.
6. Now pause the video by clicking on the Stop icon.
7. Verify the same things but this time the video should not start automatically playing when stopping dragging the handles / stopping using the arrow keys.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #9108

### CODE REVIEW

File: packages/story-editor/src/components/videoTrim/videoTrimmer.js [hunk 1]
```diff
@@ -102,6 +116,12 @@ function VideoTrimmer() {
     max: maxOffset,
     step: 100,
     minorStep: 10,
+    onPointerDown: () => setIsDraggingHandles(true),
+    onPointerUp: () => setIsDraggingHandles(false),
```
Comment (barklund) [inline packages/story-editor/src/components/videoTrim/videoTrimmer.js:123]: Oh, this one is just passed through `...rest`?

**Review by barklund — APPROVED** [review]
This works amazingly well! Love it!

**Comment by swissspidy** [comment]
Any chance of before/after screenshots/videos? 😇

**Comment by barklund** [comment]
> Any chance of before/after screenshots/videos? 😇

Done! 🐰

**Pull request merged**